### PR TITLE
Feature/create lib

### PR DESCRIPTION
**Pull request by ndoriomedov**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Build & Test ImageSupport
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the develop branch
```
Comment (BelehovEgor) [inline .github/workflows/main.yml]: comment and line 7 told me about different things how I understand

File: .vs/VSWorkspaceState.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "ExpandedNodes": [
```
Comment (BelehovEgor) [inline .vs/VSWorkspaceState.json:2]: what is it?

File: LT.DigitalOffice.ImageSupport.nuspec [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+      Previous versions:
+
+    </releaseNotes>
+    <copyright>Copyright © LT Digital Office</copyright>
+    <tags>DigitalOffice</tags>
+    <dependencies>
+      <group targetFramework="net5.0">
+        <dependency id="SixLabors.ImageSharp" version="1.0.4"/>
+        <dependency id="LT.DigitalOffice.Kernel" version="1.1.45"/>
```
Comment (BelehovEgor) [inline LT.DigitalOffice.ImageSupport.nuspec]: when a version of kernel will be greater than this, what will we do?

File: src/ImageSupport/Helpers/ContentValidationHelper.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        return true;
+      }
+      catch
+      {
+        return false;
+      }
+    }
+
+    private bool OtherFormatsResizeCheck(string content)
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ContentValidationHelper.cs]: What do others mean?

File: src/ImageSupport/Helpers/ContentValidationHelper.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    {
+      try
+      {
+        byte[] byteString = Convert.FromBase64String(content);
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ContentValidationHelper.cs]: you don't need variable here

File: src/ImageSupport/Helpers/ContentValidationHelper.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/ImageSupport/Helpers/ContentValidationHelper.cs]: why resize if you don't resize

File: src/ImageSupport/Helpers/ContentValidationHelper.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      catch
+      {
+        return false;
+      }
+    }
+    #endregion
+
+    public bool ContentCheck(string content, string extension)
+    {
+      if (extension == ".svg")
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ContentValidationHelper.cs]: maybe better create const for "svg"

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      try
+      {
+        byte[] byteString = Convert.FromBase64String(inputBase64);
+
+        using (var ms = new MemoryStream(byteString))
+        {
+          SvgDocument svgDocument = SvgDocument.Open<SvgDocument>(ms);
+          Drawing.Bitmap image = svgDocument.Draw();
+
+          if (image.Width > 150 || image.Height > 150)
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs]: create const for this (150)

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+          {
+            double maxSize = Math.Max(image.Width, image.Height);
+
+            int ratio = Convert.ToInt32(Math.Ceiling(maxSize / 150));
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs]: why ratio should be int? why not double?

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+            Drawing.Bitmap newImage = new Drawing.Bitmap(image.Width / ratio, image.Height / ratio);
+            Drawing.Graphics.FromImage(newImage).DrawImage(image, 0, 0, image.Width / ratio, image.Height / ratio);
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs:41]: why you cut the left upper angle of the picture? why not center?

File: src/ImageSupport/Helpers/ContentValidationHelper.cs [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+using LT.DigitalOffice.ImageSupport.Helpers.Interfaces;
+using SixLabors.ImageSharp;
+using Svg;
+using System;
+using System.IO;
+using Drawing = System.Drawing;
+
+namespace LT.DigitalOffice.ImageSupport.Helpers
+{
+  public class ContentValidationHelper : IContentValidationHelper
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ContentValidationHelper.cs]: why is it not IValidator?

File: src/ImageSupport/Helpers/ContentValidationHelper.cs (hunk 4 above)
Comment (ndoriomedov) [reply src/ImageSupport/Helpers/ContentValidationHelper.cs]: Others - not svg formats))

File: src/ImageSupport/Helpers/ImageResizeHelper.cs (hunk 9 above)
Comment (ndoriomedov) [reply src/ImageSupport/Helpers/ImageResizeHelper.cs:41]: It's not crop - we draw scaled picture

File: src/ImageSupport/Helpers/ImageResizeHelper.cs (hunk 8 above)
Comment (ndoriomedov) [reply src/ImageSupport/Helpers/ImageResizeHelper.cs]: Bitmap() can use only int values, image.Width and image.Height are int values too
I can make ratio double, but then i need to change at first - image.Width to double, at second - (image.Width / ratio) back to int

File: .vs/VSWorkspaceState.json (hunk 2 above)
Comment (ndoriomedov) [reply .vs/VSWorkspaceState.json:2]: something that was made by visual studio automatically, can't find it in solution explorer

File: LT.DigitalOffice.ImageSupport.nuspec (hunk 3 above)
Comment (ndoriomedov) [reply LT.DigitalOffice.ImageSupport.nuspec]: if we need, we will change it here

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+      }
+      catch
+      {
+        return (isSuccess: false, resizedContent: null, extension: extension);
+      }
+    }
+
+    #endregion
+
+    public (bool isSuccess, string resizedContent, string extension) Resize(string inputBase64, string extension)
```
Comment (ndoriomedov) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs]: will try ro make async

File: src/ImageSupport/Helpers/ContentValidationHelper.cs (hunk 10 above)
Comment (ndoriomedov) [reply src/ImageSupport/Helpers/ContentValidationHelper.cs]: added validator to kernel

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+              Drawing.ImageConverter converter = new Drawing.ImageConverter();
+
+              byteString = (byte[])converter.ConvertTo(newImage, typeof(byte[]));
+              extension = ImageFormats.png;
+
+              return (isSuccess: true,
+                resizedContent: Convert.ToBase64String(byteString),
+                extension: extension);
+            }
+            else
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs]: you can check this first and reduce code nesting

File: src/ImageSupport/Helpers/ImageResizeHelper.cs [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+          }
+        }
+        catch
+        {
+          return (isSuccess: false, resizedContent: null, extension: extension);
+        }
+      });
+    }
+
+    private Task<(bool isSuccess, string resizedContent, string extension)>  OtherFormatsResize(string inputBase64, string extension, int resizeMinValue)
```
Comment (BelehovEgor) [inline src/ImageSupport/Helpers/ImageResizeHelper.cs]: maybe better BaseResize or smth like it, i just don't know which are other

File: src/ImageSupport/ImageSupport.csproj [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <AssemblyName>LT.DigitalOffice.ImageSupport</AssemblyName>
+    <RootNamespace>LT.DigitalOffice.ImageSupport</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.47" />
+    <PackageReference Include="SixLabors.ImageSharp" Version="1.0.4" />
+    <PackageReference Include="Svg" Version="3.3.0" />
```
Comment (BelehovEgor) [inline src/ImageSupport/ImageSupport.csproj:22]: does register matter? here is Svg but in nuspec file SVG

**Review by BelehovEgor — APPROVED** [review]

File: src/ImageSupport/Helpers/ImageResizeHelper.cs (hunk 13 above)
Comment (ndoriomedov) [reply src/ImageSupport/Helpers/ImageResizeHelper.cs]: Yes, it will be better

**Review by VLiamin — APPROVED** [review]

File: src/ImageSupport/ImageSupport.csproj [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
```
Comment (MariyaSin) [inline src/ImageSupport/ImageSupport.csproj:3]: `<Version>.....</Version>`

File: .github/workflows/nuget.yml [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+        run: dotnet restore
+
+      - name: Build
+        run: dotnet build --no-restore --configuration Release --output ./artifacts
+
+      - name: Test
+        run: dotnet test --no-restore --verbosity normal
+
+      - name: Pack nuget
+        run: nuget pack LT.DigitalOffice.Kernel.nuspec -Symbols -SymbolPackageFormat snupkg -OutputDirectory ./artifacts
```
Comment (MariyaSin) [inline .github/workflows/nuget.yml]: kernel?????

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**